pkg/apis/extensions/validation/network.go

```diff
@@ -100,3 +112,42 @@ func ValidateNetworkStatusUpdate(newStatus, oldStatus *extensionsv1alpha1.Networ
 
 	return allErrs
 }
+
+var availableIPFamilies = sets.NewString(
+	string(extensionsv1alpha1.IPFamilyIPv4),
+	string(extensionsv1alpha1.IPFamilyIPv6),
+)
+
+// ValidateIPFamilies validates the given list of IP families for valid values and combinations.
+func ValidateIPFamilies(ipFamilies []extensionsv1alpha1.IPFamily, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// validate: only supported IP families
+	for i, ipFamily := range ipFamilies {
+		if !availableIPFamilies.Has(string(ipFamily)) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Index(i), ipFamily, availableIPFamilies.List()))
+		}
+	}
+
+	// validate: no duplicate IP families
+	ipFamiliesSeen := sets.NewString()
+	for i, ipFamily := range ipFamilies {
+		if ipFamiliesSeen.Has(string(ipFamily)) {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Index(i), ipFamily))
+		} else {
+			ipFamiliesSeen.Insert(string(ipFamily))
+		}
+	}
```

Comment:
Same as with core/validation.
Why is this code even duplicated?

Reply:
> Why is this code even duplicated?

Same question was answered here: https://github.com/gardener/gardener/pull/7288#discussion_r1067838733 :)

Reply:
Well, the duplication is certainly not nice, but I guess it is just another broken window ;-) .